src/Infiltrator.jl

```diff
@@ -159,57 +82,102 @@ const TEST_TERMINAL_REF = Ref{Any}(nothing)
 const TEST_REPL_REF = Ref{Any}(nothing)
 const TEST_NOSTACK = Ref{Any}(false)
 
-const EXITING = Ref{Bool}(false)
-const DISABLED = Set()
+mutable struct ScratchPad
+  store::Module
+  exiting::Bool
+  disabled::Set
+end
+function Base.show(io::IO, s::ScratchPad)
+  n = length(get_scratch_pad_names(s))
+  print(io, "Infiltrator scratch pad with $(n) name$(n == 1 ? "" : "s")")
+end
+function Base.getproperty(sp::ScratchPad, s::Symbol)
+  s === :store && return getfield(sp, :store)
+  s === :exiting && return getfield(sp, :exiting)
+  s === :disabled && return getfield(sp, :disabled)
+
+  if isdefined(sp.store, s)
+    getproperty(sp.store, s)
+  else
+    throw(UndefVarError(s))
+  end
+end
+Base.propertynames(s::ScratchPad) = keys(get_scratch_pad_names(s))
 
-const SCRATCH_PAD = Ref{Module}()
+"""
+    store
 
+Global scratch pad for storing values while `@infiltrate`ing or `@exfiltrate`ing.
 """
-    clear_disabled()
+const store = ScratchPad(Module(), false, Set())
+
+"""
+    @with ex
+
+Evaluates the expression `ex` in the context of the global scratch pad.
+
+Mainly intended for interactive use, as changes to the scratch pads state will not
+propagate into the returned expression.
+"""
+macro with(ex)
```

Comment:
Can we rename this something else to avoid a conflict with DataFramesMeta or StaticModules? 

I've filed an issue to for re-naming in DataFramesMeta [here](https://github.com/JuliaData/DataFramesMeta.jl/issues/252) because it's bad practice to squat on such a general name. But at the same time this is also a specialized implementation so it also deserves a `@withstore` or similar.